chore(docs): Revise TO documentation (#502)
mozilla/ecosystem-platform: docs/reference/team-processes/triage-process.md

- Keep ongoing bugs and issues as well as contributor PRs and assistance in Matrix top-of-mind

## I am the Triage Owner: How should I prioritize my Sprint?

:::tip
If you're new to owning triage, make sure you have access to Bugzilla (FxA) or Stripe (SubPlat).

To access the FxA Admin Panel, set up your VPN ([see instructions on Confluence][mana-vpn]) and request to be added to the appropriate [LDAP groups][bugzilla-common-scenarios-ldap].
:::

### FxA

When planning your workload for the sprint, consider taking on around 50% less than usual in order to allocate a portion of each day to triage ownership duties. If you’re in the middle of high priority feature work, fill your availability with that work. Otherwise, favor taking on a high priority maintenance item(s).

### SubPlat

The triage owner should begin each day by checking whether there are any new urgent issues in Jira, Sentry, Stripe, and Grafana, as well as questions in the subscription-platform channel.

If you are currently working on an issue and there are new S1 or S2 issues, re-evaluate whether the new issue(s) should take higher priority.

> If there are multiple S1s and S2s of seemingly equal importance and help is needed, inform clewis and/or the subplat-team channel.

Reviewer: ```suggestion
If there are multiple S1s and S2s of seemingly equal importance and help is needed, inform your manager and team in the subplat-team channel.
```

In general, the triage owner should be consistently checking whether there are new urgent or severe issues throughout the day, and be prepared to switch focus and/or redirect issues.

## What are the Triage Owner Responsibilities?

At a high level, each team's triage owner is responsible for hosting the weekly triage meeting, in which new issues in Jira and Sentry are triaged as a team.

If necessary, feel free to rely on the team's general consensus of the urgency of new issues, either in the triage meeting or async in Slack.

Any issues picked up as Triage Owner which land within a Sprint boundary should be added to that Sprint in Jira. This sets clear expectations on when a ticket is available for testing in Staging and Production across Engineering and QA.

:::note
If an error, issue, work request, etc is reported to our team in Slack, the first person to see and respond should file a bug in Jira and add the link to the Slack conversation. It will then be triaged normally in the triage meeting.
:::

### FxA

In addition, the FxA triage owner is also responsible for:

- Triaging new issues in FxA Bugzilla (as a team if there is time in the triage meeting)

> - Checking if there's any PRs, from team members or contributors, that don't have a reviewer, that you can help move along

Author: Updated to:

Checking if there are any PRs (from team members or contributors) without a reviewer that you can help move along

- Helping with Dependabot PRs
- Assisting with any high priority bugs that come up during the sprint
- Hanging out in the #fxa Matrix room

### SubPlat

In addition, the SubPlat triage owner is also responsible for:

> - Grooming the Maintenance backlog

Author: It's noted that all links are in the calendar invite of the triage meeting. If we are to add a link, it should be within the invite.

- Asynchronous triage activities

> - File bugs for concerning Subscriptions-related issues identified in review of Stripe Webhooks Events, Sentry,and Payments success/failures metrics

Author: Updated to:

File bugs for concerning Subscriptions-related issues identified in review from Stripe webhook events, Sentry, and Payments success/failure metrics

> - File bugs for concerning Subscriptions-related issues identified in review of Stripe Webhooks Events, Sentry,and Payments success/failures metrics

Reviewer 2: Nice!

- Triage new tickets for high priority requests or bugs, picking up the highest priority items as deemed appropriate
- Picking up tickets from the top of the Maintenance and/or Bug backlogs

## Host the Triage Meeting

FxA and SubPlat have separate triage meetings - optional for team members, required for triage owners. To discuss each issue and make decisions as a group, the triage owner will typically share their screen in Zoom with the team while going through Jira, Sentry, Grafana, and Bugzilla (FxA only) or Stripe (SubPlat only).

The most up-to-date links can be found in the triage meeting calendar invite. Prior to the meeting, it is recommended to pull up the links to get an idea of what you’ll be discussing and to make sure you don’t need to re-authenticate.

### Jira

> In the "FxA & Subscription Platform Triage" Board, select Backlog in the sidebar navigation, and collapse the “To Do” section to see the “Backlog” section for untriaged issues. Note, there is a _Not Subscription Platform_ filter to help target issues.

Author: You're not in the correct board. If you follow the updated instructions, you will see the To Do section to collapse and filter.

<img width="239" alt="Screenshot 2024-01-30 at 9 52 26 AM" src="https://github.com/mozilla/ecosystem-platform/assets/28129806/4d3b5446-59e5-4d32-a28c-a23d0616d853">

> In the "FxA & Subscription Platform Triage" Board, select Backlog in the sidebar navigation, and collapse the “To Do” section to see the “Backlog” section for untriaged issues. Note, there is a _Not Subscription Platform_ filter to help target issues.

Reviewer 2: Thank you! I had to logout and log back in. That's so weird. I did select the correct board but Backlog took me back to the previous board.

No need to change anything then. 👍 I still think a list is more favourable than commas, but that's IMHO.

When reviewing issues, leave tickets that are not applicable to your team as-is since those tickets will be triaged in the associated triage meeting. The following should be reviewed for each ticket:

> - **Make sure the issue is clear and ready to be worked on.** If more info or context is needed that an engineer in the triage meeting can provide, either leave a summary comment of what’s stated (if it’s quick) or ask them to comment on it. You can also @-mention those who may be able to answer pending questions. If more info is needed and the issue seems high priority, consider pinging the reporter directly about the issue after the triage meeting.

Reviewer 2: Nit: add comma after "context is needed"

Q: for my own self, what do you mean by "quick"?
Q: how do we determine "high priority" (can we link to where priorities are broken down)?

Author: > Q: for my own self, what do you mean by "quick"?
I think if you're able to gather the extra context to leave in a comment, then do so. If not, ask them to leave a comment.

> Q: how do we determine "high priority" (can we link to where priorities are broken down)?
It will be mentioned during the triage meeting.

Tagging @LZoog , as she is the initial author of this section, in case I didn't interpret it correctly.

- **Check that the issue type is correct.** The default type when filing in Jira is “task”. Other types include “spike” (for research task or work breakdown task) and “bug” (in which you will need to fill out the “Found in” (environment) and “[Severity][bug-severity]” (S1 is highest, S4 is lowest) fields).
- **Add an initial estimated number of story points**, see [Estimation and Point Values][point-estimation].
- **Add the appropriate epic, labels, and/or issue links, if applicable.**
  - It’s uncommon for a new issue in triage to lack an epic if it should be associated with one, though it’s good to be aware of existing epics and the roadmap.

>   - Add the appropriate labels (e.g. `needsdiscussion`, `needslegal`, `needsproduct`, `needsux`). Other common labels to consider are `maintenance` and `qa-` (see [labels][labels-we-use] for more).

Reviewer 2: Nit: no issue here other than I hardly see these labels applied. I'd maybe consider examples of the common ones we come across. Otherwise, clarify that this would be a team decision and not on the shoulders of the triage owner?

Author: I think if I clarified it for this point, I would also have to clarify it for every point on this list as well. Since this is the "Host the Triage Meeting" section (and from our TO experience), the team will help the triage owner fill any missing information in the ticket before it is moved out of the backlog.

  - Issue links create a link between issues (e.g., issues can be “blocked by X issue”, “relates to X issue”, and more).
- **Move the issue out of the backlog section.** If the issue seems high priority, it can either be added to the current or following sprint by updating the “Sprint” field, or moved to the top of the “To Do” list by right-clicking and following the context menu. If someone on the call asks to be assigned, fill out the Assignee field. Finally, update the status from "New" to "To Do".

### Sentry

> We use Sentry to capture errors that occur across the entire stack. While we should monitor our production instances most closely, we should also keep an eye out on staging instances for new issues exposed by QA. There are many projects in the FxA group on Sentry and while it’s always helpful to poke through more than what’s listed here, we generally check the following deployments (check the triage calendar invite for links):

Reviewer 2: Nit: add comma after "Sentry" for readability

Suggestion: update "check the triage calendar" to "see triage calendar" to eliminate double use of "check" in the same sentence

If you see a bug filed requesting [OAuth client details][oauth-integration], post in the FxA team Slack channel. Reference the bug, and bring it to the attention of SRE.

### Stripe Triage (SubPlat only)

> Triaging Stripe consists of ensuring that failed webhook event deliveries are successful upon retries in production.

Reviewer 2: Nit: update "production" to "Production"
